processing/src/main/java/io/druid/query/extraction/NamespacedExtraction.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.extraction;
+
+import com.fasterxml.jackson.annotation.JacksonInject;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Function;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.inject.name.Named;
+import com.metamx.common.StringUtils;
+
+import javax.annotation.Nullable;
+import javax.validation.constraints.NotNull;
+import java.nio.ByteBuffer;
+
+/**
+ * Namespaced extraction is a special case of DimExtractionFn where the actual extractor is pulled from a map of known implementations.
+ * In the event that an unknown namespace is passed, a simple reflective function is returned instead.
```

Comment:
Can you expand on the logic for why referring to a namespace that doesn't exist is allowed?

The best argument I can come up with is so that you can drop a namespace and then redefine it, but I think we can support that just by making the code able to swap out namespaces...